packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -161,6 +161,13 @@ class CupertinoPicker extends StatefulWidget {
 
 class _CupertinoPickerState extends State<CupertinoPicker> {
   int _lastHapticIndex;
+  FixedExtentScrollController _controller;
+
+  @override
+  void initState() {
+    super.initState();
+    _controller = widget.scrollController ?? FixedExtentScrollController();
+  }
```

Comment:
You'd need didUpdateWidget here too no?

Reply:
Added didUpdateWidget and a test case to confirm that this works properly